src/lib/crypto/secp256k1.ts

```diff
@@ -549,6 +619,40 @@ const wrapSecp256k1Wasm = (
         : false
       : false;
 
+  const signMessageHashRecoverable = (privateKey: Uint8Array,messageHash: Uint8Array) => {
+    fillMessageHashScratch(messageHash);
+    return withPrivateKey<RecoverableSignature>(privateKey, () => {
+      if (secp256k1Wasm.signRecoverable(contextPtr, internalRSigPtr, messageHashScratch, privateKeyPtr) !== 1) {
+        throw new Error('Failed to sign message hash. The private key is not valid.');
+      }
+      secp256k1Wasm.recoverableSignatureSerialize(contextPtr,sigScratch,recoveryNumPtr,internalRSigPtr);
+      // tslint:disable-next-line:no-object-literal-type-assertion no-angle-bracket-type-assertion
+      return <RecoverableSignature>{
```

Comment:
On the `no-angle-bracket-type-assertion` – the newer way to tell TypeScript about a type is `YOUR_OBJECT as RecoverableSignature`. Then you won't need that tslint line disable.

Reply:
Fair enough. Looks like I'll still need the `no-object-literal-type-assertion` though.